src/service/randoService.js

```diff
@@ -0,0 +1,106 @@
+var logger = require("../log/logger");
+
+module.exports = {
+  findAllChoosers (randos) {
+    var choosers = [];
+
+    if (randos) {
+      for (var i = 0; i < randos.length; i++) {
+        if (!this.isRandoWasChooser(randos[i], randos)) {
+          choosers.push(randos[i]);
+        }
+      }
+    }
+
+    return choosers;
+  },
+  //PRIVATE?
+  isRandoWasChooser (rando, randos) {
+    logger.trace("[randoService.putRandoToUserAsync.cleanBucket.isRandoWasChooser]", "Process rando ", rando.randoId, " for ", randos.length, " randos");
+    for (var i = 0; i < randos.length; i++) {
+      if (rando.randoId === randos[i].strangerRandoId) {
+        logger.trace("[randoService.putRandoToUserAsync.cleanBucket.isRandoWasChooser]", "Rando ", rando.randoId, " can NOT be chooser");
+        return true;
+      }
+    }
+    logger.trace("[randoService.putRandoToUserAsync.cleanBucket.isRandoWasChooser]", "Rando ", rando.randoId, " can be chooser");
+    return false;
+  },
+  isRandoFullyExchanged (rando, randos) {
+    if (!rando || !randos) {
+      return false;
+    }
+
+    return this.isRandoWasChooser (rando, randos) 
+      && rando.strangerRandoId 
+      && this.isRandoWasChooser(this.findRandoByRandoId(rando.strangerRandoId, randos), randos);
+  },
+  //PRIVATE?
+  findRandoByRandoId(randoId, randos) {
```

Comment:
Can we change randos from array to map:
randoId -> rando

So it will be more lightweight from algorithm point of view?